cluster/cluster.go

```diff
@@ -15,4 +38,85 @@ func Init(name, version string, primary bool, started time.Time) {
 		primaryChange: time.Now(),
 		stateChange:   time.Now(),
 	}
+	Mode = m
+	go probePeers(interval)
+}
+
+func Stop() {
+	close(shutdown)
+}
+
+func AddPeer(remoteAddr *url.URL) {
+	mu.Lock()
+	peer := &Node{
+		remoteAddr:  remoteAddr,
+		stateChange: time.Now(),
+	}
+	peers = append(peers, peer)
+	go peer.Probe()
+	mu.Unlock()
+}
+
+func GetPeers() []*Node {
+	mu.Lock()
+	p := make([]*Node, len(peers))
+	copy(p, peers)
+	mu.Unlock()
+	return p
+}
+
+func probePeers(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	for {
+		select {
+		case <-shutdown:
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			mu.Lock()
+			for _, peer := range peers {
+				go peer.Probe()
```

Comment:
The HTTP call is made without the lock being held.  So the easiest solution here is to just track if a probe is currently underway and if it is, immediately return.